### DISCUSSION TITLE
How mc mirror --watch work

### DISCUSSION THREAD

**Discussion by sangk54** [minio/minio#18963] [Q&A] [answered]
I use minio as a hot-tier storage at separate local servers. I want to real-time replicate data from minio to s3. 
Because destination is s3, I can't use server-side bucket replication; instead I have to use mc mirror
```
mc mirror --watch minio/bucket s3/bucket
```
I have the following two questions:
1. I only want to sync the source files that are added or updated, not sync the deleted files.
2. I'm wondering how mc identifies files that are added or updated. Does mc use the API to traverse all files on the MinIO server, or does the MinIO server have a notification mechanism to inform the client about changes?

**Comment by harshavardhana** [comment, marked as answer]
> * I only want to sync the source files that are added or updated, not sync the deleted files.

That's by default if you didn't provide --remove with --watch



> 2\. I'm wondering how mc identifies files that are added or updated. Does mc use the API to traverse all files on the MinIO server, or does the MinIO server have a notification mechanism to inform the client about changes?

notifications via a HTTP connection

**Reply by sangk54** [reply]
Thank for your reply.
1. I tested the command `mc mirror --watch source dest`. If I delete a file on the source, mc will listen to this event and delete the corresponding file on the destination.

2. Does "mc mirror --watch" ensure that all added files will be copied on the destination. For example, in case of temporary network loss,... I need a highly reliable synchronization mechanism. Do I need to write a job that listens to Kafka ObjectCreated events and copies the file to S3?
